tests/benchmarks/MicroPerf/Collections.fs

```diff
@@ -0,0 +1,298 @@
+    [<Benchmark>]
+    member x.ListUpdateAtBeginning() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.updateAt 0 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ListUpdateEnd() =
+        async {
+            List.init x.Length (fun _ -> 0)
+            |> List.updateAt (x.Length - 1) 1
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    /// Array
+    [<Benchmark>]
+    member x.ArrayRemoveAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeAt 0 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayRemoveAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeAt (x.Length - 1) 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayRemoveManyAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeManyAt 0 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayRemoveManyAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.removeManyAt (x.Length - 11) 10 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayInsertAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.insertAt 0 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayInsertAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.insertAt (x.Length - 1) 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayInsertManyAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.insertManyAt 0 [1..10] 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayInsertManyAtEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.insertManyAt (x.Length - 11) [1..10]
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    [<Benchmark>]
+    member x.ArrayUpdateAtBeginning() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.updateAt 0 1 
+            |> ignore
+        }
+        |> Async.StartAsTask
+
+    [<Benchmark>]
+    member x.ArrayUpdateEnd() =
+        async {
+            Array.init x.Length (fun _ -> 0)
+            |> Array.updateAt (x.Length - 1) 1
+            |> ignore
+        }
+        |> Async.StartAsTask
+        
+    /// Seq
+    [<Benchmark>]
+    member x.SeqRemoveAtBeginning() =
+        async {
+            Seq.init x.Length (fun _ -> 0)
+            |> Seq.removeAt 0 
```

Comment:
For Seq I think it's particularly important to only measure the operation - it may be `Seq.init` is doing more allocation that we don't want to measure.

So I think use `Seq.removeAt` etc. on a fixed, pre-allocated list or array

Reply:
I also changed the Seq.toList to Seq.iter ignore, hopefully, that'll be more accurate.